cli/command/manifest/util.go

```diff
@@ -0,0 +1,179 @@
+package manifest
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/cli/cli/manifest/fetcher"
+	"github.com/docker/docker/pkg/homedir"
+)
+
+type osArch struct {
+	os   string
+	arch string
+}
+
+// dirOpenError
+type dirOpenError struct {
+}
+
+func (e dirOpenError) Error() string {
+	return "cannot perform open on a directory"
+}
+
+// Remove any unsupported os/arch combo
+// list of valid os/arch values (see "Optional Environment Variables" section
+// of https://golang.org/doc/install/source
+// Added linux/s390x as we know System z support already exists
+var validOSArches = map[osArch]bool{
+	{os: "darwin", arch: "386"}:      true,
```

Comment:
We should probably pare down that list. I'm open to whatever the community decides it should be. They're not all supported.